src/core/shape_inference/include/unsqueeze_shape_inference.hpp

```diff
@@ -63,7 +63,11 @@ std::vector<TRShape> shape_infer(const Unsqueeze* op,
             }
         }
     } else {
-        out_shape = ov::PartialShape::dynamic();
+        if (arg_shape.rank().is_static() && input_shapes[1].is_static() && input_shapes[1].size() == 1) {
+            out_shape = ov::PartialShape::dynamic(arg_shape.rank() + input_shapes[1][0].get_length());
```

Comment:
we can only do this for single axis, in case there are many axes -- they have to be uniquified. Please see the code above.
I propose to change check to spot shape {1} and {} -- in such case we will be sure the axis is unique and output rank would be equal to `input_rank + 1`